unix_integration/pam_kanidm/src/pam/mod.rs

```diff
@@ -286,10 +286,16 @@ impl PamHooks for PamKanidm {
                     };
                     match prompt.cred_type {
```

Comment:
Where are we creating this cred_type to pass back up to the pam module here?

Reply:
That's happening in the idprovider. The idprovider is calling `passwd_prompt` at unix_integration/src/unix_proto.rs:59.